components/camel-ai/camel-langchain4j-web-search/src/test/java/org/apache/camel/component/langchain4j/web/search/LangChain4jGoogleWebSearchEngineIT.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.langchain4j.web.search;
+
+import java.util.List;
+
+import dev.langchain4j.web.search.WebSearchEngine;
+import dev.langchain4j.web.search.google.customsearch.GoogleCustomWebSearchEngine;
+import org.apache.camel.CamelContext;
+import org.apache.camel.Exchange;
+import org.apache.camel.test.junit5.CamelTestSupport;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.EnabledIfEnvironmentVariable;
+
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+// Instruction to generate one https://developers.google.com/custom-search/v1/overview
+@EnabledIfEnvironmentVariable(named = "GOOGLE_API_KEY", matches = ".*")
+// Instructions to create a Google search engine https://developers.google.com/custom-search/docs/tutorial/creatingcse
+@EnabledIfEnvironmentVariable(named = "GOOGLE_SEARCH_ENGINE_ID", matches = ".*")
+public class LangChain4jGoogleWebSearchEngineIT extends CamelTestSupport {
+    public static final String WEB_SEARCH_URI = "langchain4j-web-search:googleTest";
+
+    @Override
+    protected CamelContext createCamelContext() throws Exception {
+        CamelContext context = super.createCamelContext();
+
+        WebSearchEngine googleSearchEngine = GoogleCustomWebSearchEngine.builder()
+                .apiKey(System.getenv("GOOGLE_API_KEY"))
+                .csi(System.getenv("GOOGLE_SEARCH_ENGINE_ID"))
+                .logRequests(true)
+                .logResponses(true)
+                .build();
+
+        context.getRegistry().bind("web-search-engine", googleSearchEngine);
+
+        return context;
+    }
```

Comment:
This API is becoming more restricted with 4.8 and, in the future, 4.9. I'd suggest using 

`protected void bindToRegistry(Registry registry) throws Exception {`

instead.

Reply:
oh thanks

Reply:
The BindToRegistry annotation is cool

Reply:
I improved that part. Let me know if that looks good to you